gitbutler-ui/src/lib/components/CommitDialog.svelte

```diff
@@ -39,31 +40,55 @@
 	export let cloud: ReturnType<typeof getCloudApiClient>;
 	export let user: User | undefined;
 	export let selectedOwnership: Writable<Ownership>;
+	export const expanded = persisted<boolean>(false, 'commitBoxExpanded_' + branch.id);
 
 	const aiGenEnabled = projectAiGenEnabled(projectId);
 	const runCommitHooks = projectRunCommitHooks(projectId);
 	const currentCommitMessage = projectCurrentCommitMessage(projectId, branch.id);
-	export const expanded = persisted<boolean>(false, 'commitBoxExpanded_' + branch.id);
 
-	let commitMessage: string = get(currentCommitMessage) || '';
-	let isCommitting = false;
-	let textareaElement: HTMLTextAreaElement;
-	$: if (textareaElement && commitMessage && expanded) {
-		textareaElement.style.height = 'auto';
-		textareaElement.style.height = `${textareaElement.scrollHeight + 2}px`;
+	function getCommitMessageTitleAndDescription(commitMessage: string) {
+		// Split the commit message into title and description
+		// get the first line as title and the rest as description
+		const [summary, ...description] = commitMessage.split('\n');
+		return {
+			summary: summary || '',
+			// remove top new line
+			description: description.join('\n').replace(/^\n/, '') || ''
+		};
+	}
+
+	function concatCommitMessage(message: { summary: string; description: string }) {
```

Comment:
This looks a bit weird to me, can this function take two regular parameters called summary and description?

Reply:
Yes, will fix

Reply:
Working on these fixes now. I added an object as a parameter here because `commitMessageSet` is an object, so it's easier just to pass the whole object like this `concatCommitMessage(commitMessageSet)`